### ISSUE TITLE
@Serializable on enum should not break compilation

### ISSUE THREAD

**Issue by qwwdfsad** [Kotlin/kotlinx.serialization#304]
Adding `@Serializable` to enum breaks compilation:
```
private interface Letter
@Serializable
private enum class Letters : Letter { A, B, C }
```

Such contract from the plugin is inconvenient, especially without meaningful diagnostic.
Marking enum with `@Serializable` should not change anything

**Comment by markGilchrist** [comment]
@sandwwraith 
where can I find an example of a `still can be serialized using custom companion object.`

**Comment by sandwwraith** [comment]
@markGilchrist Here: https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/custom_serializers.md#customizing ?

**Comment by sandwwraith** [comment]
Fixed in 1.3.30
